ports/stm/common-hal/rgbmatrix/RGBMatrix.c

```diff
@@ -34,20 +34,23 @@
 extern void _PM_IRQ_HANDLER(void);
 
 void *common_hal_rgbmatrix_timer_allocate() {
-    return stm_peripherals_find_timer();
+    TIM_TypeDef * timer = stm_peripherals_find_timer();
+    stm_peripherals_timer_reserve(timer);
+    return timer;
 }
 
 void common_hal_rgbmatrix_timer_enable(void* ptr) {
-    HAL_NVIC_EnableIRQ(TIM6_DAC_IRQn);
+    TIM_TypeDef *tim = (TIM_TypeDef*)ptr;
+    HAL_NVIC_EnableIRQ(stm_peripherals_timer_get_irqnum(tim));
 }
 
 void common_hal_rgbmatrix_timer_disable(void* ptr) {
     TIM_TypeDef *tim = (TIM_TypeDef*)ptr;
     tim->DIER &= ~TIM_DIER_UIE;
-    HAL_NVIC_DisableIRQ(TIM6_DAC_IRQn);
 }
 
 void common_hal_rgbmatrix_timer_free(void* ptr) {
+    TIM_TypeDef *tim = (TIM_TypeDef*)ptr;
+    stm_peripherals_timer_free(tim);
     common_hal_rgbmatrix_timer_disable(ptr);
```

Comment:
not sure which order these two should go in, or whether calling timer_disable is redundant now.

Reply:
I think it's fine, you're just more explicit about calling the `tim->DIER &= ~TIM_DIER_UIE` than the other peripherals. I'll revisit this when I try and fix the flickering issue.